### PR TITLE
[PHI] Move oneDNN helper classes to new location

### PR DESCRIPTION
**Pull request by piotrekobi**
### PR types

Others
### PR changes

Others
### Describe

- Moves most classes from paddle/fluid/platform/mkldnn_reuse.h to paddle/phi/kernels/funcs/onednn/onednn_reuse.h.
- Changes "mkldnn" to "onednn" in function, class and variable names in PHI files wherever possible

These changes will make it easier to convert oneDNN kernels to PHI, because no more changes should need to be made to the onednn_reuse.h file.

### CODE REVIEW

File: paddle/fluid/framework/new_executor/data_transfer.cc [hunk 1]
```diff
@@ -212,8 +212,8 @@ std::shared_ptr<OperatorBase> TransferLayout(const std::string& var_name,
     out_layout = framework::DataLayout::kNCHW;
   }
 
-  if (in_layout == framework::DataLayout::MKLDNN &&
-      out_layout != framework::DataLayout::MKLDNN) {
+  if (in_layout == framework::DataLayout::ONEDNN &&
+      out_layout != framework::DataLayout::ONEDNN) {
     auto target_layout = phi::OneDNNContext::tls().get_cur_paddle_data_layout();
     VLOG(4) << "TransDataLayoutFromMKLDNN: " << in_layout << "->"
```
Comment (Silv3S) [inline paddle/fluid/framework/new_executor/data_transfer.cc]: `TransDataLayoutFromMKLDNN` -> `TransDataLayoutFromOneDNN`

**Review by Silv3S — DISMISSED** [review]
LGTM

**Comment by piotrekobi** [comment]
@chenwhql @jczaja Please review & merge

**Review by jczaja — APPROVED** [review]
LGTM

**Pull request merged**